src/contexts/TodoContext.tsx

```diff
@@ -0,0 +1,231 @@
+import React, {
+  useState, useEffect,
+} from 'react';
+import { TodoContextType } from '../types/TodoContextType';
+import { Todo } from '../types/Todo';
+import { TypeOfFiltering } from '../types/TypeOfFiltering';
+import { ErrorType } from '../types/ErrorType';
+import { client } from '../utils/fetchClient';
+import { UserWarning } from '../UserWarning';
+
+export const TodoContext = React.createContext<TodoContextType>({
+  todos: [],
+  tempTodo: null,
+  changeData: () => {},
+  deleteData: () => {},
+  filterType: TypeOfFiltering.All,
+  setFilterType: () => {},
+  dataError: '',
+  setError: () => {},
+  Error: () => {},
+  addTodo: () => {},
+  editTodo: -1,
+  setEditTodo: () => {},
+  changeTodo: () => {},
+  activeLoader: [],
+  shouldFocus: true,
+  setInputValue: () => {},
+  inputValue: '',
+  setTodos: () => {},
+});
+
+type Props = {
+  children: React.ReactNode,
+};
+
+const USER_ID = 9940;
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState<TypeOfFiltering>(
+    TypeOfFiltering.All,
+  );
+  const [dataError, setError] = useState<string>('');
+  const [activeLoader, setActiveLoader] = useState<number[]>([]);
+  const [shouldFocus, setShouldFocus] = useState<boolean>(true);
+  const [editTodo, setEditTodo] = useState<number>(-1);
+  const [inputValue, setInputValue] = useState('');
+
+  let timeoutId: ReturnType<typeof setTimeout>;
+
+  const Error = (error: ErrorType) => {
+    setError(error);
+
+    timeoutId = setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  const getTodos = () => {
+    return client.get<Todo[]>(`/todos?userId=${USER_ID}`);
+  };
+
+  const fetchData = async () => {
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+    } catch {
+      Error(ErrorType.Load);
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, []);
```

Comment:
![2024-03-12_11h20_10](https://github.com/mate-academy/react_todo-app-with-api/assets/61904995/55ac72d2-2d4f-4ea9-b079-8a0fc91c9daa)
```suggestion
  

  useEffect(() => {
  const fetchData = async () => {
    try {
      const todosFromServer = await getTodos();

      setTodos(todosFromServer);
    } catch {
      Error(ErrorType.Load);
    }
  };
  
    fetchData();
  }, []);
```